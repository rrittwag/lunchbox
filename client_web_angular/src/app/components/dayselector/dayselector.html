<div class="day-control">
  <div class="row horizontal-align">
    <div class="col-xs-12 col-sm-9 col-md-6 col-lg-4">
      <div class="row vertical-align">
        <div class="col-xs-2">
          <a class="btn btn-primary glyphicon glyphicon-chevron-left"
              ng-click="goPrevDay()" ng-disabled="!hasPrevDay()"></a>
        </div>
        <div class="col-xs-8 text-center">
          <h2><small>{{selectedDay | formatToWeekday}}</small></h2>
          <h2>{{selectedDay | date:'dd.MM.yyyy'}}</h2>
        </div>
        <div class="col-xs-2">
          <a class="btn btn-primary glyphicon glyphicon-chevron-right pull-right"
              ng-click="goNextDay()" ng-disabled="!hasNextDay()"></a>
        </div>
      </div>
    </div>
  </div>
</div>
