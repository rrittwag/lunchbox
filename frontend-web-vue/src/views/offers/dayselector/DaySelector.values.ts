export enum DaySelectorDirection {
  PREVIOUS = 'prev',
  NEXT = 'next',
}

export const LABEL_GO_TO_PREVIOUS_DAY = 'Zu vorherigem Tag wechseln'
export const LABEL_GO_TO_NEXT_DAY = 'Zu n√§chstem Tag wechseln'
