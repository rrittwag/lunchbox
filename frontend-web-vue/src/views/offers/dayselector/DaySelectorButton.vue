<script setup lang="ts">
import {
  DaySelectorDirection,
  LABEL_GO_TO_NEXT_DAY,
  LABEL_GO_TO_PREVIOUS_DAY,
} from '@/views/offers/dayselector/DaySelector.values'
import AngleLeftIcon from '~icons/fa-solid/angle-left'
import AngleRightIcon from '~icons/fa-solid/angle-right'

const props = defineProps<{
  disabled?: boolean
  direction: DaySelectorDirection
}>()
const emit = defineEmits<{
  (e: 'click'): void
}>()

const onClick = () => emit('click')

const isPrevious = props.direction === DaySelectorDirection.PREVIOUS
const title = isPrevious ? LABEL_GO_TO_PREVIOUS_DAY : LABEL_GO_TO_NEXT_DAY
</script>

<template>
  <button
    class="cursor-pointer origin-center text-primary-500 hover:scale-110 hover:text-primary-600 disabled:cursor-not-allowed disabled:text-primary-400 disabled:opacity-25"
    :class="{ 'active:text-primary-800': !props.disabled }"
    :title="title"
    :aria-label="title"
    :disabled="props.disabled"
    :aria-keyshortcuts="isPrevious ? 'ArrowLeft' : 'ArrowRight'"
    @click="onClick"
  >
    <AngleLeftIcon v-if="isPrevious" class="h-16 w-16" aria-hidden="true" />
    <AngleRightIcon v-else class="h-16 w-16" aria-hidden="true" />
  </button>
</template>
