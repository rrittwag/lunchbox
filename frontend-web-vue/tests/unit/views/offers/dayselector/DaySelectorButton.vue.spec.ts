import { mount } from '@vue/test-utils'
import DaySelectorButton from '@/views/offers/dayselector/DaySelectorButton.vue'
import { DaySelectorDirection } from '@/views/offers/dayselector/DaySelectorDirection'

describe('DaySelectorButton', () => {
  it('renders snapshot for previous day', () => {
    const wrapper = mount(DaySelectorButton, {
      props: { direction: DaySelectorDirection.PREVIOUS },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders snapshot for next day', () => {
    const wrapper = mount(DaySelectorButton, {
      props: { direction: DaySelectorDirection.NEXT },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('has no `active` style when disabled', () => {
    const wrapper = mount(DaySelectorButton, {
      props: { direction: DaySelectorDirection.PREVIOUS, disabled: true },
    })

    // Trotz disabled-Zustand kann ein Button bei Klick active werden.
    // -> https://stackoverflow.com/a/12592035
    // Für diesen unschönen Fall unterbinden wir das Stylen des Buttons.
    expect(wrapper.classes().find((c) => c.startsWith('active'))).toBeUndefined()
  })

  it('emits click event', async () => {
    const wrapper = mount(DaySelectorButton, {
      props: { direction: DaySelectorDirection.PREVIOUS },
    })

    await wrapper.find('button').trigger('click')

    expect(wrapper.emitted().click).toBeDefined()
    expect(wrapper.emitted().click!.length).toBe(1)
  })
})
