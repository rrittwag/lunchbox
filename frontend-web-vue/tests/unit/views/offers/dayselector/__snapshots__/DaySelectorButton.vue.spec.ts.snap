// Vitest Snapshot v1

exports[`DaySelectorButton > renders snapshot for next day 1`] = `
<button
  aria-keyshortcuts="ArrowRight"
  aria-label="Zu nächstem Tag wechseln"
  class="origin-center text-primary-500 hover:scale-110 hover:text-primary-600 disabled:cursor-not-allowed disabled:text-primary-400 disabled:opacity-25 active:text-primary-800"
  title="Zu nächstem Tag wechseln"
>
  <svg
    aria-hidden="true"
    class="h-16 w-16"
    height="1.2em"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 256 512"
    width="1.2em"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4l-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
    />
  </svg>
</button>
`;

exports[`DaySelectorButton > renders snapshot for previous day 1`] = `
<button
  aria-keyshortcuts="ArrowLeft"
  aria-label="Zu vorherigem Tag wechseln"
  class="origin-center text-primary-500 hover:scale-110 hover:text-primary-600 disabled:cursor-not-allowed disabled:text-primary-400 disabled:opacity-25 active:text-primary-800"
  title="Zu vorherigem Tag wechseln"
>
  <svg
    aria-hidden="true"
    class="h-16 w-16"
    height="1.2em"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 256 512"
    width="1.2em"
  >
    <path
      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
      fill="currentColor"
    />
  </svg>
</button>
`;
